README.md
# Thème Facyl


## Installation

```bash
npm i 
npx husky install 
npx husky add .husky/commit-msg 'npx commitlint --edit $1'
```

Si besoin, installation de la CLI de Gulp:
```bash
npm install --global gulp-cli
```

Pour générer les certificats pour le https en local:
```bash
sudo apt install mkcert
mkdir -p /var/www/certs && cd /var/www/certs && mkcert localhost 192.168.x.xxx ::1
```


## Debuggers

Air-light comes with PHP_CodeSniffer for PHP files, stylelint for SCSS/CSS files and eslint for JS files built inside gulpfile.js. Please note, you need to configure global versions of these separately!

It's also recommended to use Query Monitor plugin, as some debugging messages goes straight to it's logger.

### For gulp
PHP_CodeSniffer needs to be installed under /usr/local/bin/phpcs with WordPress-Coding-Standards for php-debuggers to work properly in gulp. If you don't want to use phpcs with gulp, you can disable it by commenting out or deleting line gulp.watch(phpSrc, ['phpcs']);.

The golden rule here is to make sure the commands stylelint, eslint and phpcs work from command line.

**How to install for Gulp**

```bash
mkdir -p ~/Projects && cd ~/Projects && git clone -b master --depth 1 https://github.com/squizlabs/PHP_CodeSniffer.git phpcs
git clone -b master https://github.com/PHPCompatibility/PHPCompatibility
git clone -b master --depth 1 https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards.git wpcs
```

**Please note: Replace yourusername name with your actual user name!**
```bash
sudo ln -s /home/yourusername/Projects/phpcs/bin/phpcs /usr/local/bin/phpcs
sudo chmod +x /usr/local/bin/phpcs
```

**Please note: Replace yourusername name with your actual user name!**
```bash
phpcs --config-set installed_paths "/home/yourusername/Projects/wpcs","/home/yourusername/Projects/PHPCompatibility"
```

Test your standards with phpcs -i, it should display something like this:

```bash
The installed coding standards are PEAR, Zend, PSR2, MySource, Squiz, PSR1, PSR12, PHPCompatibility, WordPress, WordPress-Extra, WordPress-Docs and WordPress-Core
```

```bash
npm i stylelint eslint -g
```

Check that other linters work:
```bash
stylelint -v 
eslint -v
```

### For your editor

It's also best to have all stylelint, eslint, phpcs living inside your editor. We think Visual Studio Code is best for this, check out the plugins inside vscode-settings repository to make sure everything is installed.

## Convention de commit
type(scope?): subject #scope is optional; multiple scopes are supported (current delimiter options: "/", "" and ",")

Types list by convention :

- build: changes that affect the build system or external dependencies
- ci: changes to our CI configuration files and scripts
- docs: documentation only changes.
- feat: used when a commit adds a new feature to your application or library.
- fix: used when a commit represents a bug fix for your application.
- perf: a code change that improves performance.
- refactor: a code change that neither fixes a bug nor adds a feature.
- revert: revert code changes made before.
- style: changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc).
- test: adding missing tests or correcting existing tests.
- chore: other task.

---

## Astuces code css
` @include button(max-content,var(--color-black),var(--color-black),var(--color-white),var(--color-white),30px,1px solid var(--color-black),1px solid var(--color-black),var(--font-weight-semibold),true);`

CHANGELOG.md
### 9.1.7: 2022-09-28

* Add external-link-icon styles for mobile
* Remove setup_editor_styles function because it breaks stage+prod and block styles still load correctly without that
* Improve anchors.js formatting
* Add a11y-skip-link JS module for ensuring that skip links always focus to first heading

### 9.1.6: 2022-09-14

* Fix font-size not inheriting to Gutenberg editor
* Fix rich text font size in Gutenberg editor
* Fix Gutenberg editor font-sizing issues
* Deprecate responsive-font() mixin in favor of clamp-calc() function

### 9.1.5: 2022-09-12

* Fix: Use ACF defined font styles in ACF fields
* Fix acf icons visibility issue
* Fix link colors in sidebar
* Add default font-family for form select items
* Remove opionated label font-weight
* Remove opionated nav default dropshadow from sub menus
* Remove opionated nav default border-color from sub menus
* Remove opionated bubble tip from sub menus

### 9.1.4: 2022-09-07

* Fix: Viewport padding/white-space on mobile devices
* Fix: Leftover variable not defined

### 9.1.3: 2022-09-07

* Consistency for variable `--font-weight-paragraph`
* Remove `has-light-bg` and `has-dark-bg` classes that are no longer used by back to top feature
* Increase padding on mid-sized screens to prevent container sticking to sides

### 9.1.2: 2022-09-07

* Clean up is-external-link from CSS that is no longer used
* Consistency in variables: `--font-size-paragraphs` -> `--font-size-paragraph`
* Remove unused `--font-size-default` and combine with `--font-size-paragraph`
* Fix list font-sizes not inheriting from paragraph styles
* Define default font sizes and line-heights on body level instead in separate elements

### 9.1.1: 2022-09-06

* Improve form checkbox and radiob button styles
* Form checkboxes and radio buttons: Add bouncy check animation
* Clarify reset for checkboxes and radio buttons for gravity forms
* Combine `--line-height-paragraphs-blog` and `--line-height-paragraphs` to one unified CSS variable: `--line-height-paragraph`

### 9.1.0: 2022-09-01

* Improve post title color and size accessibility in Gutenberg editor
* Remove unnecessary float reset from nav item
* Add image-background-layer to image helper classes
* Fix enumerating grid in columns
* Bump tested up WordPress version to 6.0.2

### 9.0.9: 2022-08-25

* Add custom version of sanitize.css
* Improve external-links.js JavaScript
* Ignore external link arrows for links that have only imgs in them
* Add ignore classes for external links
* Generate SVG arrow for external links via JS instead of CSS
* Clean up old different versions of external-link.svg files

### 9.0.8: 2022-08-23

* Deprecate lazyload from CSS as it is no longer needed after loading="lazy"
* Add image-background helper class for imgs as backgrounds
* Add CSS clamp function as helper for responsive font sizes
* Add CSS aspect-ratio helper for responsive images

### 9.0.7: 2022-08-22

* Add box-model helper
* Remove leftover getLocalization declaration
* Remove block-file-slug.svg placeholder, replace with .gitkeep
* Improvement: Consistent $transition-duration, less movement

### 9.0.6: 2022-06-30

* Upgrade to [devpackages 2.4.7](https://github.com/digitoimistodude/devpackages/releases/tag/2.4.7)
* Remove air_helper_custom_settings_post_ids hook that is no longer used (kudos to latenssi @ wpfi Slack for spotting it!)

### 9.0.5: 2022-06-09

* Prevent mobile navigation flickering before JS has been loaded

### 9.0.4: 2022-06-09

* Add default localization for accessible carousels to support Air-blocks 1.1.2

### 9.0.3: 2022-06-08

* Remove leftover %screen-reader-text from editor styles
* Move editor styles outside of the global gutenberg-editor-styles.scss
* Fix form elements in ACF block previews
* A11y: Localizations for possible carousel blocks from Air-blocks

### 9.0.2: 2022-05-27

* Fix critical regression about ACF blocks not outputting, Fixes #153

### 9.0.1: 2022-05-26

* Disable dropdown box-shadow by default
* Improve placeholder for new project
* Fix typo in CSS variable name
* Fix critical JS error with calling getChildAltText() in vain
* Fix outdated release command

### 9.0.0: 2022-05-24

#### Major update

* Rebranding: Update default logo
* Deprecate %screen-reader-text, use mixin instead
* Deprecate old lazy load styles from future themes
* Add var(--box-shadow-sub-menu) for desktop navigation
* Add more clear starting point with demo content placeholder
* Add support for anchors in the block (thanks @timiwahalahti)
* Disable support for additional CSS classes in blocks (thanks @timiwahalahti)
* Bump tested up WordPress version to 6.0
* Fix back to top label for screen readers
* Add data-version attribute to footer indicating facyl version in use

### 8.4.7: 2022-05-24

* Remove _slick.scss, deprecate slick-carousel slider (carousel block is part of [air-blocks](https://github.com/digitoimistodude/air-blocks) now)

### 8.4.6: 2022-05-23

* Combine media queries
* Deprecate heading-hero component

### 8.4.5: 2022-05-20

* Use only one burger animation, revert burger style on mobile

### 8.4.4: 2022-05-20

* Simplify external link selector
* Add js-trigger to skip-link
* Combine nav-mobile media queries
* Update stylelintrc rules block-closing-brace-newline-after and at-rule-empty-line-before
* Fix cubic-bezier mixin $duration variable
* Fix iOS 15 nav width style bug
* Fix iOS 15 scrolling bug with `height: -webkit-fill-available;`
* New navigation animation: slide (default), change to fade by setting `$nav-slide-animation: false;`

### 8.4.3: 2022-05-13

* Fixes #134, nav-toggle menu position shifting on iOS 15.1

### 8.4.2: 2022-05-12

* Add anchors as JS module
* Deprecate vanilla-lazyload and support for older JS-based lazy loading methods
* Drop jQuery
* Rewrite accessible back to top functionality and styles
* Combine external link labels with external link JS module
* Refactor front-end.js

### 8.4.1: 2022-05-12

* Fix paddings on editor
* Remove deprecated text columns block

functions.php
<?php
/**
 * Gather all bits and pieces together.
 * If you end up having multiple post types, taxonomies,
 * hooks and functions - please split those to their
 * own files under /inc and just require here.
 *
 * @Date: 2019-10-15 12:30:02
 * @Last Modified by:   Roni Laukkarinen
 * @Last Modified time: 2022-09-28 14:36:25
 *
 * @package facyl
 */

namespace Air_Light;

/**
 * The current version of the theme.
 */
define( 'AIR_LIGHT_VERSION', '9.1.7' );

// We need to have some defaults as comments or empties so let's allow this:
// phpcs:disable Squiz.Commenting.InlineComment.SpacingBefore, WordPress.Arrays.ArrayDeclarationSpacing.SpaceInEmptyArray

/**
 * Theme settings
 */
add_action( 'after_setup_theme', function() {
  $theme_settings = [
    /**
     * Theme textdomain
     */
    'textdomain' => 'facyl',

    /**
     * Image and content sizes
     */
    'image_sizes' => [
      'small'   => 300,
      'medium'  => 700,
      'large'   => 1200,
    ],
    'content_width' => 800,

    /**
     * Logo and featured image
     */
    'default_featured_image'  => null,
    'logo'                    => '/svg/logo.svg',
    'logo_linkedin'           => '/svg/linkedin.svg',
    'logo_insta'              => '/svg/insta.svg',
    'chevron_right'           => '/svg/chevron-right.svg',
    'chevron_up'              => '/svg/chevron-up.svg',

    /**
     * Custom setting group settings when using Air setting groups plugin.
     * On multilingual sites using Polylang, translations are handled automatically.
     */
    'custom_settings' => [
      // 'your-custom-setting' => [
      //   'id' => Your custom setting post id,
      //   'title' => 'Your custom setting',
      //   'block-editor' => true,
      //  ],
    ],

    'social_media_accounts'  => [
      // 'twitter' => [
      //   'title' => 'Twitter',
      //   'url'   => 'https://twitter.com/digitoimistodude',
      // ],
    ],

    /**
     * All links are checked with JS, if those direct to external site and if,
     * indicator of that is included. Exclude domains from that check in this array.
     */
    'external_link_domains_exclude' => [
      'localhost:3000',
      // 'airdev.test',
      // 'airwptheme.com',
      'localhost',
    ],

    /**
     * Menu locations
     */
    'menu_locations' => [
      'primary' => __( 'Primary Menu', 'facyl' ),
      'footer' => __( 'Footer Menu', 'facyl' ),
      'confidentiality' => __( 'Confidentialité Menu', 'facyl' ),
    ],

    /**
     * Taxonomies
     *
     * See the instructions:
     * https://github.com/digitoimistodude/facyl#custom-taxonomies
     */
    'taxonomies' => [
      // 'your-taxonomy' => [
      //   'name' => 'Your_Taxonomy',
      //   'post_types' => [ 'post', 'page' ],
      // ],
    ],

    /**
     * Post types
     *
     * See the instructions:
     * https://github.com/digitoimistodude/facyl#custom-post-types
     */
    'post_types' => [
      // 'your-post-type' => 'Your_Post_Type',
    ],

    /**
     * Gutenberg -related settings
     */
    // Register custom ACF Blocks
    'acf_blocks' => [
          [
            'name'            => 'eco-conception',
            'title'           => 'Bloc Eco Conception',
            'render_template' => '/template-parts/blocks/eco-conception.php',
            'category'        => 'formatting',
            'enqueue_script'  => '/sass/template-parts/_eco-conception.scss',
          ],
      // [
      //   'name'           => 'block-file-slug',
      //   'title'          => 'Block Visible Name',
      //   // You can safely remove lines below if you find no use for them
      //   'prevent_cache'  => false, // Defaults to false,
      //   // Icon defaults to svg file inside svg/block-icons named after the block name,
      //   // eg. svg/block-icons/block-file-slug.svg
      //   //
      //   // Icon setting defines the dashicon equivalent: https://developer.wordpress.org/resource/dashicons/#block-default
      //   // 'icon'  => 'block-default',
      // ],
    ],

    // Custom ACF block default settings
    'acf_block_defaults' => [
      'category'          => 'facyl',
      'mode'              => 'auto',
      'align'             => 'full',
      'post_types'        => [
        'page',
      ],
      'supports'  => [
        'align'           => false,
        'anchor'          => true,
        'customClassName' => false,
      ],
      'render_callback'   => __NAMESPACE__ . '\render_acf_block',
    ],

    // Restrict to only selected blocks
    // Set the value to 'all' to allow all blocks everywhere
   'allowed_blocks' => [
      'default' => [
      ],
      'post' => [
        'core/archives',
        'core/audio',
        'core/buttons',
        'core/categories',
        'core/code',
        'core/column',
        'core/columns',
        'core/coverImage',
        'core/embed',
        'core/file',
        'core/freeform',
        'core/gallery',
        'core/heading',
        'core/html',
        'core/image',
        'core/latestComments',
        'core/latestPosts',
        'core/list',
        'core/more',
        'core/nextpage',
        'core/paragraph',
        'core/preformatted',
        'core/pullquote',
        'core/quote',
        'core/block',
        'core/separator',
        'core/shortcode',
        'core/spacer',
        'core/subhead',
        'core/table',
        'core/textColumns',
        'core/verse',
        'core/video',
      ],
    ],

    // If you want to use classic editor somewhere, define it here
    'use_classic_editor' => [],

    // Add your own settings and use them wherever you need, for example THEME_SETTINGS['my_custom_setting']
    'my_custom_setting' => true,
  ];

  $theme_settings = apply_filters( 'air_light_theme_settings', $theme_settings );

  define( 'THEME_SETTINGS', $theme_settings );
} ); // end action after_setup_theme

/**
 * Required files
 */
require get_theme_file_path( '/inc/hooks.php' );
require get_theme_file_path( '/inc/includes.php' );
require get_theme_file_path( '/inc/template-tags.php' );
require get_theme_file_path( '/inc/clean.php' );

// Run theme setup
add_action( 'init', __NAMESPACE__ . '\theme_setup' );
add_action( 'after_setup_theme', __NAMESPACE__ . '\build_theme_support' );
